src/Module/OStatus/PortableContacts.php

```diff
@@ -0,0 +1,278 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * @see https://web.archive.org/web/20160405005550/http://portablecontacts.net/draft-spec.html
+ */
+
+namespace Friendica\Module\OStatus;
+
+use Friendica\App;
+use Friendica\BaseModule;
+use Friendica\Content\Text\BBCode;
+use Friendica\Core\Cache\Capability\ICanCache;
+use Friendica\Core\Config\Capability\IManageConfigValues;
+use Friendica\Core\L10n;
+use Friendica\Core\Protocol;
+use Friendica\Core\System;
+use Friendica\Database\Database;
+use Friendica\Module\Response;
+use Friendica\Network\HTTPException;
+use Friendica\Util\DateTimeFormat;
+use Friendica\Util\Profiler;
+use Psr\Log\LoggerInterface;
+
+class PortableContacts extends BaseModule
+{
+	/** @var IManageConfigValues */
+	private $config;
+	/** @var Database */
+	private $database;
+	/** @var ICanCache */
+	private $cache;
+
+	public function __construct(ICanCache $cache, Database $database, IManageConfigValues $config, L10n $l10n, App\BaseURL $baseUrl, App\Arguments $args, LoggerInterface $logger, Profiler $profiler, Response $response, array $server, array $parameters = [])
+	{
+		parent::__construct($l10n, $baseUrl, $args, $logger, $profiler, $response, $server, $parameters);
+
+		$this->config   = $config;
+		$this->database = $database;
+		$this->cache    = $cache;
+	}
+
+	protected function rawContent(array $request = [])
+	{
+		if ($this->config->get('system', 'block_public') || $this->config->get('system', 'block_local_dir')) {
+			throw new HTTPException\ForbiddenException();
+		}
+
+		if ($this->args->getArgc() > 1) {
+			// Only the system mode is supported
+			throw new HTTPException\NotFoundException();
+		}
+
+		$format = $request['format'] ?? '' ?: 'json';
```

Comment:
I never really understand what you're doing here tbh *gg*

- if `$request['format']` is null, set it to `''`
- **afterwards**, if it's empty, set it to `'json'`

What intention is behind it? What case do you have in mind?
For me, it would rather be easier to have
```suggestion
		$format = $request['format'] ?? ''json';
```

Reply:
`$var ?? '' ?: $else` is a direct translation of `!empty($var) ? $var : $else`. the catch-all aspect of `empty()` isn't always warranted, but I've been erring on the same side when I have not been able to evaluate the merits of each individual case.

It looks like it can be simplified in this case because it's a string, but other cases with integers can be more tricky.